src/asynk/async_worker_pool.rs

```diff
@@ -51,8 +47,9 @@ where
     }
 
     #[async_recursion]
-    pub async fn supervise_worker(queue: AsyncQueue<Tls>) -> Result<(), Error> {
-        let result = Self::run_worker(queue.clone()).await;
+    pub async fn supervise_worker(queue: AQueue) -> Result<(), Error> {
+        let q: &'static mut AQueue = &mut queue.clone();
```

Comment:
we have a borrowing error here , but if we solve them i think this will be the way to generic this

Reply:
we will think later